src/main/java/org/folio/linked/data/service/resource/ResourceMarcServiceImpl.java

```diff
@@ -81,43 +78,89 @@ private void validateMarkViewSupportedType(Resource resource) {
       return;
     }
     throw new ValidationException(
-      "Resource is not supported for MARC view",
-      "type", resource.getTypes().stream()
-      .map(ResourceTypeEntity::getUri)
-      .collect(Collectors.joining(", ", "[", "]"))
+      "Resource is not supported for MARC view", "type",
+      resource.getTypes().stream()
+        .map(ResourceTypeEntity::getUri)
+        .collect(Collectors.joining(", ", "[", "]"))
     );
   }
 
-  private Long createResource(Long incomingId, String incomingSrsId, Resource mapped) {
-    logMarcAction(incomingId, incomingSrsId, "not found by id and srsId", "created");
-    return saveAndPublishEvent(mapped, ResourceCreatedEvent::new);
+  private Long createResource(Resource resource) {
+    logMarcAction(resource, "not found by id and srsId", "be created");
+    return saveAndPublishEvent(resource, ResourceCreatedEvent::new);
+  }
+
+  private Long replaceResource(Resource resource) {
+    if (resource.isAuthority()) {
+      return replaceAuthority(resource);
+    }
+    return replaceBibliographic(resource);
```

Comment:
FYI - We do not have to worry about `replaceBibliographic` per the new workflow.
We have to consider only authority records via this kafka flow per the new approach. We will create a new ticket to remove unwanted logic from this class.

Reply:
oh, that's news. Ok, thank you for the info. Let's discuss it also